contracts/HubPool.sol

```diff
@@ -270,19 +287,17 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         // Decrement the unclaimedPoolRebalanceLeafCount.
         refundRequest.unclaimedPoolRebalanceLeafCount--;
 
-        // Transfer the bondAmount to back to the proposer, if this was not done before for this refund bundle.
-        if (!refundRequest.proposerBondRepaid) {
-            refundRequest.proposerBondRepaid = true;
+        // Transfer the bondAmount to back to the proposer, if this the last executed leaf. Only sending this once all
+        // leafs have been executed acts to force the data worker to execute all bundles or they wont receive their bond.
+        if (refundRequest.unclaimedPoolRebalanceLeafCount == 0)
             bondToken.safeTransfer(refundRequest.proposer, bondAmount);
-        }
```

Comment:
it's better to only pay the relayer back their bond one all bundles are executed.